models/src/main/kotlin/motif/models/Scope.kt

```diff
@@ -80,6 +80,10 @@ private class ScopeFactory(
     }
 
     private fun visit(scopeClass: IrClass) {
+        if (scopeClass.typeArguments.isNotEmpty()) {
```

Comment:
Parsing errors should be thrown inside the instantiation of `ValidScope`. Make this check in the `ValidScope` init block.